addons/mail/static/src/core/web/chatter.js

```diff
@@ -157,24 +157,15 @@ export class Chatter extends Component {
             this.scrollPosition.restore();
         });
         onPatched(this.scrollPosition.restore);
-        onWillUpdateProps((nextProps) => {
-            this.load(nextProps.threadId, ["followers", "attachments", "suggestedRecipients"]);
-            if (nextProps.threadId === false) {
-                this.state.composerType = false;
-            }
-            this.attachmentUploader.thread = this.threadService.getThread(
-                nextProps.threadModel,
-                nextProps.threadId
-            );
-            if (this.onNextUpdate) {
-                if (!this.onNextUpdate(nextProps)) {
-                    this.onNextUpdate = null;
-                }
-            }
-            if (nextProps.threadId) {
-                this.closeSearch();
+        useBus(this.env.bus, "MAIL:RELOAD-CHATTER", ({ detail }) => {
+            // Only reload chatter if the model is the same.
+            if (detail.model === this.props.threadModel) {
```

Comment:
Why not? Isn't it because `load` is broken and doesn't take model as param?
If that is the case, load should be adapted to take model as param as well, and condition removed.

Reply:
I think the problem it is not always allowed to fetch `detail.model` with `detail.id` due to security restrictions (as if you remove the code you will see fails when it is trying to read some `random record` because `onWillLoadRoot` is triggered every time there is a form opened, or records cannot be a `thread`) and also it only makes sense to fetch data with the same model. So I think the code should be kept.